lib/Sema/CSSimplify.cpp

```diff
@@ -4774,6 +4834,46 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
                                       flags, locatorB);
     };
 
+    // Check if any property wrappers on the base of the member lookup have
+    // mactching members that we can fall back to.
+    if (auto dotExpr =
+            dyn_cast_or_null<UnresolvedDotExpr>(locator->getAnchor())) {
+      auto baseExpr = dotExpr->getBase();
+      auto resolvedOverload = getResolvedOverloadSets();
+      while (resolvedOverload) {
+        if (resolvedOverload->Locator->getAnchor() == baseExpr)
+          break;
+        resolvedOverload = resolvedOverload->Previous;
+      }
+
+      if (resolvedOverload && resolvedOverload->Choice.isDecl()) {
+        if (auto *decl =
+                dyn_cast<VarDecl>(resolvedOverload->Choice.getDecl())) {
+          if (decl->hasAttachedPropertyWrapper()) {
+            auto rawWrapperTy = decl->getAttachedPropertyWrapperType(0);
+            auto wrapperTy =
+                openUnboundGenericType(rawWrapperTy, resolvedOverload->Locator);
```

Comment:
Here is one more suggestion - instead of opening wrapper type here and then adding constraint to match it with value type, what you can do is form `BoundGenericType` from wrapper type and pass already known value type as it's generic argument.